### PR TITLE
Add `irc_puppet_prejoin_commands`

### PR DESCRIPTION
**Pull request by llmII**
This adds support for interpolating the current nick of a connection so as to remove the need for hardcoded modes being set in the code. Provides for a separte list for the main connection (listener or the simple mode relay) and for puppets.

We discussed something akin to cascading previously of the prejoin onto the puppets where puppets would execute both sets of commands but the listener would only execute prejoin. I think it's best to keep these separate completely since I know inspircd allows for configurations where a user can set a mode on themself but cannot remove it. A possibly good example is mode +x being locked after set some places. So if the listener sets +x due to prejoin, and the connections set +x as well due to, but you want to -x later in the puppet_prejoin... you're stuck.

This hasn't been tested yet, and should supplant #70. This does not set any defaults whatsoever (perhaps a sane default of +D for puppet_prejoin makes sense?).

### CODE REVIEW

**Comment by qaisjp** [comment]
Thanks! Could we rename the `irc_prejoin_commands` to `irc_listener_prejoin_commands`?

**Comment by llmII** [comment]
Will do, commit in a few minutes.

**Comment by llmII** [comment]
Done.

File: README.md [hunk 1]
```diff
@@ -66,31 +66,32 @@ The binary takes three flags:
...
+| `puppet_username`              | No               | username of discord account being puppeted     | Yes                          | username to connect to irc with                                                                                                                                          |
+| `webirc_pass`                  | No               |                                                | Yes                          | optional, but recommended for regular (non-simple) usage. this must be obtained by the IRC sysops                                                                        |
+| `irc_prejoin_commands`         | Yes              |                                                | Yes                          | list of commands for the listener IRC connection to execute (right before joining channels)                                                                                      |
```
Comment (qaisjp) [inline README.md:83]:
```suggestion
| `irc_listener_prejoin_commands`| Yes              |                                                | Yes                          | list of commands for the listener IRC connection to execute (right before joining channels)                                                                                      |
```

**Review by qaisjp — APPROVED** [review]
Thank you!

**Pull request merged**